mlir/lib/Bindings/Python/IRAttributes.cpp

```diff
@@ -708,8 +843,8 @@ class PyDenseElementsAttribute
         std::string message;
         llvm::raw_string_ostream os(message);
         os << "Expected a static ShapedType for the shaped_type parameter: "
-           << py::repr(py::cast(*explicitType));
-        throw py::value_error(message);
+           << nb::cast<std::string_view>(nb::repr(nb::cast(*explicitType)));
```

Comment:
The temporary result of py::repr is kept alive through the operator application, right? (I usually avoid this idiom because then I don't have to think about c++ temporary lifetime extension rules)

Reply:
(you do this a lot so just confirming the once here)

Reply:
Ok, that's usually how I end up back at this point. Since these are all exception messages and presumably not performance critical (and also don't always get the best testing), let's be conservative/safe with std::string, unless if one of the c++ gurus which can sometimes be found in these parts says it is ok.

Reply:
Done.